RSHighscores.body.php

```diff
@@ -69,6 +69,42 @@ private static function retrieveHiscores( $hs, $player ) {
 	}
 
 	/**
+	 * Lookup hiscores data from object cache before retrieving from the site.
+	 *
+	 * @param string $hs Which hiscores API to retrieve from.
+	 * @param string $player Player's display name.
+	 * @return string Raw hiscores data
+	 */
+	private static function lookupHiscores( $hs, $player ) {
+		// Instance of the object cache to retrieve the hiscores data from.
+		$objCache = wfGetCache( CACHE_ANYTHING );
+
+		// Try to retrieve the hiscores data from the object cache.
+		$data = $objCache->get( 'rshiscores-' . $player . '-' . $hs );
+
+		// Couldn't find in the object cache, so retrieve from the site.
+		if ( $data === false ) {
+			// If not blocked from too many requests or technical issues, then lookup.
+			if ( $objCache->get( 'rshiscores-blocked') === false ) {
+				$data = self::retrieveHiscores( $hs, $player );
+
+				// Request failed, so no requests for 15 min.
+				if ( $data == 'C28' ) {
```

Comment:
Apparently, the GED API returns a 200 status with either no text or a warning of too many requests in too little time. I'll look into running some sort of test to actually find out what happens, but maybe we could add something that checks for it being something that resembles a CSV file?